tests/integration_tests/lightgbm_tuner_tests/test_optimize.py

```diff
@@ -923,3 +923,44 @@ def test_optuna_callback(self) -> None:
             tuner._tune_params(["num_leaves"], 10, optuna.samplers.TPESampler(), "num_leaves")
 
         assert callback_mock.call_count == 10
+
+    def test_get_best_booster(self) -> None:
+        unexpected_value = 20  # out of scope.
+
+        params = {"verbose": -1, "lambda_l1": unexpected_value}  # type: Dict
+        dataset = lgb.Dataset(np.zeros((10, 10)))
+
+        study = optuna.create_study()
+        with TemporaryDirectory() as tmpdir:
+            tuner = LightGBMTunerCV(
+                params,
+                dataset,
+                study=study,
+                model_dir=tmpdir,
+                return_cvbooster=True,
+            )
+            with mock.patch.object(_OptunaObjectiveCV, "_get_cv_scores", return_value=[1.0]):
+                tuner.tune_regularization_factors()
+
+            best_booster = tuner.get_best_booster()
+            for booster in best_booster.boosters:
+                assert booster.params["lambda_l1"] != unexpected_value
+
+        tuner2 = LightGBMTuner(params, dataset, valid_sets=dataset, study=study)
+
+        # Resumed study does not have the best booster.
+        with pytest.raises(ValueError):
+            tuner2.get_best_booster()
```

Comment:
This test case is not related to `LightGBMTunerCV`. Could you remove it?
BTW, I think we can get the best booster if we specify `model_dir`. What do you think?

```python
        with TemporaryDirectory() as tmpdir:
            ...
            tuner2 = LightGBMTunerCV(params, dataset, study=study, model_dir=tmpdir, return_cvbooster=True)
            best_booster = tuner2.get_best_booster()
            for booster in best_booster.boosters:
                assert booster.params["lambda_l1"] != unexpected_value
```